azure-quantum/azure/quantum/workspace.py

```diff
@@ -131,111 +114,97 @@ def __init__(
         resource_id: Optional[str] = None,
         location: Optional[str] = None,
         credential: Optional[object] = None,
-        user_agent: Optional[str] = None,
+        connection_string: Optional[str] = None,
```

Comment:
Did this change from the spec review, Yi? @04diiguyi , I thought we had to do something like Workspace().from_connection_string("<connection_string>"): 
https://azure.github.io/azure-sdk/python_design.html#azure-sdk-api-design
![image](https://github.com/microsoft/azure-quantum-python/assets/20933017/a4a9f8c4-2cbc-4541-9352-a25aa2267181)

Reply:
Yes. @vxfield , let's sync up on this. Please follow the python sdk rule we have in this PR. https://github.com/microsoft/azure-quantum-python/pull/556/commits/3cf130002f2d5058c0715a7158bc551a667719f1